Native image builds failing with dependencies from private repository

### What happened?

I've been testing an integration that relies on dependencies from a private maven repository. The integration works as expected in non-native mode however when running in native mode, the builder pod is unable to resolve the external dependencies. It's _almost_ as if the builder pod is using different maven settings and is then unable to authenticate to the repo, but this is just a suspicion since the integration works when building within the operator.

### Steps to reproduce

1. Setup camel-k 2.1.0 in a cluster with maven settings that have access to a private repository
2. Create an integration that includes a dependency from the private repository
3. Run the integration in native build mode
4. Integration should fail while attempting to resolve the private dependencies

### Relevant log output

```shell
Logs from the builder pod running the native image build:

quarkus-native Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/camel/quarkus/camel-quarkus-groovy-dsl/3.2.0/camel-quarkus-groovy-dsl-3.2.0.jar (4.8 kB at 1.1 kB/s)
quarkus-native [INFO] ------------------------------------------------------------------------
quarkus-native [INFO] BUILD FAILURE
quarkus-native [INFO] ------------------------------------------------------------------------
quarkus-native [INFO] Total time:  33.819 s
quarkus-native [INFO] Finished at: 2023-11-02T15:33:18Z
quarkus-native [INFO] ------------------------------------------------------------------------
quarkus-native [ERROR] Failed to execute goal on project camel-k-integration: Could not resolve dependencies for project org.apache.camel.k.integration:camel-k-integration:jar:2.1.0: The following artifacts could not be resolved: com.backbase.gc.generic:grandcentral-kamelet-generic:jar:1.0.9-SNAPSHOT, com.backbase.grandcentral:grandcentral-connectors-generic:jar:1.0.11-SNAPSHOT-29: Could not find artifact com.backbase.gc.generic:grandcentral-kamelet-generic:jar:1.0.9-SNAPSHOT -> [Help 1]
quarkus-native [ERROR] 
quarkus-native [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
quarkus-native [ERROR] Re-run Maven using the -X switch to enable full debug logging.
quarkus-native [ERROR] 
quarkus-native [ERROR] For more information about the errors and possible solutions, please read the following articles:
quarkus-native [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```


### Camel K version

v2.1.0

--- a/pkg/builder/runtime_support.go
+++ b/pkg/builder/runtime_support.go
@@ -63,7 +63,8 @@ type NativeAdapter struct {
 
 // BuildCommands -- .
 func (n *NativeAdapter) BuildCommands() string {
-	return "cd " + n.Directory() + " && ./mvnw package -Dquarkus.package.type=native --global-settings settings.xml"
+	// We must override the local repo as it's not shared from the builder container
+	return "cd " + n.Directory() + " && ./mvnw $(cat MAVEN_CONTEXT) package -Dquarkus.package.type=native -Dmaven.repo.local=./repo"
 }
 
 // Directory -- .

